### PR TITLE
Keras TBCallback: Make batch-level logging configurable

### PR DESCRIPTION
**Pull request by lgeiger**
4921064dd535d84aa031f8116e583b151dd46e97 introduced batch-level metrics logging.

This can lead to performance problems in certain situations, especially with fast training models. This will result in the following warning to be triggered:
```python-traceback
WARNING:tensorflow:Method on_batch_end() is slow compared to the batch update. Check your callbacks.
```

This PR makes this behavior configurable by introducting the `batch_level_metrics` config option to allow users to restore the old behavior.

@fchollet Would you prefer adding a more fine grained option that makes the frequency of logging batch-level metrics during training configurable, similar to `histogram_freq`?

### CODE REVIEW

**Comment by caisq** [comment]
@lgeiger Does it make sense to add a unit test for this?

**Comment by lgeiger** [comment]
At the current state I think almost everything is covered by the existing unit tests. Though if we decide to add a more elaborate config option (i.e. configure the logging frequency) I'm happy to add more unit tests.

File: tensorflow/python/keras/callbacks.py [hunk 1]
```diff
@@ -723,6 +725,7 @@ class TensorBoard(Callback):
   def __init__(self,
                log_dir='./logs',
                histogram_freq=0,
+               batch_level_metrics=True,
```
Comment (fchollet) [inline tensorflow/python/keras/callbacks.py]: Should this default to False?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. Yes, I think adding a frequency argument would make sense here.

**Comment by lgeiger** [comment]
@fchollet Thanks for the review. I added a frequency argument so the logging frequency can be configured as well as unit tests.

**Comment by lgeiger** [comment]
@fchollet, @caisq Friendly ping 😇. Any change this can be reviewed?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by lgeiger** [comment]
> Nagging Assignee @caisq: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

In the meantime this behavior was introduced in c98ffffcb4e0cc668c0ff7b73d51677a7eb7dcf4.

This is now my second PR (after #20753) that went obsolete during the time it took to get a review. Are there any discussions (GH issues) going on around improving the management of contributions from outside Google? I'd be happy to join the conversation.
